boot/bootutil/src/loader.c

```diff
@@ -903,26 +904,16 @@ boot_copy_image_to_sram(struct boot_loader_state *state, int slot,
         return BOOT_EFLASH;
     }
 
-    /*
-     * Scorpio flash driver only handles a single sector at a time.
-     * TODO: Let scorpio handle infinite length.
-     */
-#define SCORP_SECTOR_SZ 512
-
 #ifdef CONFIG_SCORPIO_BOOTLOADER
     /* Direct copy from flash to its new location in SRAM. */
     BOOT_LOG_INF("");
     BOOT_LOG_INF("Copying image from %s slot into LPDDR", slot ? "SECONDARY" : "PRIMARY");
     
-    int local_offset = 0;
-    while (img_sz > 0) {
-        int sz = MIN(SCORP_SECTOR_SZ, img_sz);
-        rc = flash_area_read(fap_src, local_offset, (void *)(uint64_t)(img_dst + local_offset), sz);
-        if (rc != 0) {
-            BOOT_LOG_INF("Error whilst copying image from Flash to SRAM: %d", rc);
-        }
-        img_sz -= sz;
-        local_offset += sz;
+    rc = flash_area_read(fap_src, 0, (void *)img_dst, img_sz);
```

Comment:
I know this is not strictly related to this PR per se, but I've really wanted this in the past and just saw the `Copying image...` line right here in the PR review which reminded me: do you think it would be useful to print the image size and load address as part of the `Copying image...` message?
```
BOOT_LOG_INF("Copying image from %s slot into LPDDR (%u @ 0x%lx)...", slot ? "SECONDARY" : "PRIMARY", img_sz, img_dst);
```

Reply:
Sure, lemme add that.